lms/djangoapps/edraak_university/tests/tests.py

```diff
@@ -0,0 +1,480 @@
+
+        self.assertNotContains(res, '/university/id/', msg_prefix='The link should not appear')
+
+
+@ddt.ddt
+class TestHelpers(ModuleStoreTestCase):
+    """
+    Tests for the UniversityID helper functions.
+    """
+
+    def test_is_feature_enabled_helper(self):
+        with patch.dict(settings.FEATURES, EDRAAK_UNIVERSITY_APP=True):
+            self.assertTrue(helpers.is_feature_enabled(), 'Should respect the feature')
+
+        with patch.dict(settings.FEATURES, EDRAAK_UNIVERSITY_APP=False):
+            self.assertFalse(helpers.is_feature_enabled(), 'Should respect the feature')
+
+        with patch.object(settings, 'FEATURES', {}):
+            self.assertFalse(helpers.is_feature_enabled(), 'Should default to False when the feature is missing')
+
+    @ddt.unpack
+    @ddt.data(
+        {'urlconf': 'lms.urls', 'should_enable': False, 'msg': 'Should be disable when user=None in LMS'},
+        {'urlconf': 'cms.urls', 'should_enable': True, 'msg': 'Should be enabled when user=None in CMS'},
+    )
+    def test_enable_if_no_use_cms(self, urlconf, should_enable, msg):
+        with override_settings(ROOT_URLCONF=urlconf):
+            course = Mock(enable_university_id=True)
+            self.assertEquals(helpers.is_feature_enabled_on_course(course, user=None), should_enable, msg=msg)
+
+    @patch.dict(settings.FEATURES, {'EDRAAK_UNIVERSITY_APP': True, 'EDRAAK_UNIVERSITY_CSV_EXPORT': False})
+    @ddt.data(False, True)
+    def test_is_csv_export_enabled_on_course_helper_disabled(self, course_enable_university_id):
+        course = Mock(enable_university_id=course_enable_university_id)
+        is_export_enabled = helpers.is_csv_export_enabled_on_course(course)
+        self.assertFalse(is_export_enabled, 'Export should be disabled when the feature flag is')
+
+    @patch.dict(settings.FEATURES, {'EDRAAK_UNIVERSITY_APP': True, 'EDRAAK_UNIVERSITY_CSV_EXPORT': True})
+    def test_is_csv_export_enabled_on_course_helper_enabled(self):
+        self.assertFalse(
+            expr=helpers.is_csv_export_enabled_on_course(Mock(enable_university_id=False)),
+            msg='Export should be disabled when the feature flag is',
+        )
+
+        self.assertTrue(
+            expr=helpers.is_csv_export_enabled_on_course(Mock(enable_university_id=True)),
+            msg='Export should be enabled when all three features are enabled',
+        )
+
+    def test_get_university_id_helper(self):
+        course = CourseFactory.create()
+        user = UserFactory.create()
+
+        self.assertIsNone(helpers.get_university_id(AnonymousUser(), unicode(course.id)))
+
+        self.assertIsNone(helpers.get_university_id(user, unicode(course.id)))
+
+        UniversityID.objects.create(
+            course_key=unicode(course.id),
+            user=user,
+        )
+
+        self.assertIsNotNone(helpers.get_university_id(user, unicode(course.id)), 'Should not return an ID')
```

Comment:
Isn't the function testing whether get_university_id actually returns an id, so in case of failure we need to tell the developer that  **should return an ID** which is not what you are saying here